sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,374 @@
+                               "disk_io_reads", \
+                               "disk_io_writes", \
+                               "reserved_blocks"]
+
+        # These are the fields that we are interested in saving to disk to protect against
+        # reboots or crashes
+        self.statedb_json_sync_fields = self.dynamic_fields[3:7]
+
+        # Connect to STATE_DB and create Storage device table
+        self.state_db = daemon_base.db_connect("STATE_DB")
+        self.device_table = swsscommon.Table(self.state_db, STORAGE_DEVICE_TABLE)
+        
+        # Load the FSIO RW values from state_db and JSON file and reconcile latest information
+        self._load_fsio_rw_statedb()
+        self._load_fsio_rw_json()
+        self._determine_sot()
+
+    def _get_configdb_intervals(self):
+        self.config_db = daemon_base.db_connect("CONFIG_DB")
+        config_info = dict(self.config_db.hgetall('STORMOND_CONFIG|INTERVALS'))
+        self.timeout = int(config_info.get('daemon_polling_interval', STORMOND_MAIN_THREAD_SLEEP_SECS))
+        self.fsstats_sync_interval = int(config_info.get('fsstats_sync_interval', STORMOND_SYNC_TO_DISK_SECS))
+
+        self.log_info("Polling Interval set to {} seconds".format(self.timeout))
+        self.log_info("FSIO JSON file Interval set to {} seconds".format(self.fsstats_sync_interval))
+        
+
+    # Get the total and latest FSIO reads and writes from JSON file
+    def _load_fsio_rw_json(self):
+        try:
+           if not os.path.exists(FSIO_RW_JSON_FILE):
+               self.log_info("{} not present.".format(FSIO_RW_JSON_FILE))
+               return
+
+           with open(FSIO_RW_JSON_FILE, 'r') as f:
+               self.fsio_rw_json = json.load(f)
+               self.fsio_json_file_loaded = True
+
+        except Exception as e:
+            self.log_error("JSON file could not be loaded: {}".format(str(e)))
+
+        return
+
+
+    # Sync the total and latest procfs reads and writes from STATE_DB to JSON file on disk
+    def _sync_fsio_rw_json(self):
+
+        self.log_info("Syncing total and latest procfs reads and writes from STATE_DB to JSON file")
+
+        json_file_dict = {disk:{} for disk in self.storage.devices}
+        try:
+            for device in self.storage.devices:
+                for field in self.statedb_json_sync_fields:
+                    json_file_dict[device][field] = self.state_db.hget('STORAGE_INFO|{}'.format(device), field)
+            
+            self.fsio_sync_time = time.time()
+            json_file_dict["successful_sync_time"] = str(self.fsio_sync_time)
+
+            with open(FSIO_RW_JSON_FILE, 'w+') as f:
+                json.dump(json_file_dict, f)
+                self.state_db.hset("{}|{}".format(STORAGE_DEVICE_TABLE,FSSTATS_SYNC_TIME_KEY), "successful_sync_time", str(self.fsio_sync_time))
+
+        except Exception as ex:
+            self.log_error("Unable to sync state_db to disk: {}".format(str(ex)))
```

Comment:
The following error is seen few times on config reload. Can you please investigate?

```
2024 May 24 07:51:18.097242 r-leopard-42 INFO pmon#supervisord 2024-05-24 04:51:18,096 WARN stopped: pcied (exit status 143)
2024 May 24 07:51:18.097271 r-leopard-42 INFO pmon#thermalctld[54]: Caught signal 'SIGTERM' - exiting...
2024 May 24 07:51:18.097429 r-leopard-42 INFO pmon#thermalctld: Stop thermal monitoring loop
2024 May 24 07:51:18.105745 sonic INFO database#supervisord: redis 52:M 24 May 2024 04:51:18.105 * Background saving terminated with success
2024 May 24 07:51:18.112982 r-leopard-42 INFO pmon#thermalctld[54]: Shutting down with exit code 143...
2024 May 24 07:51:18.227112 r-leopard-42 INFO pmon#supervisord 2024-05-24 04:51:18,226 WARN stopped: thermalctld (exit status 143)
2024 May 24 07:51:18.227223 r-leopard-42 INFO pmon#stormond[53]: Caught signal 'SIGTERM'
2024 May 24 07:51:18.227223 r-leopard-42 INFO pmon#stormond[53]: Syncing latest procfs reads and writes to disk
2024 May 24 07:51:18.227261 r-leopard-42 INFO pmon#stormond[53]: Syncing total and latest procfs reads and writes from STATE_DB to JSON file
2024 May 24 07:51:18.227316 r-leopard-42 INFO lldp#supervisord 2024-05-24 04:51:18,226 WARN received SIGTERM indicating exit request
2024 May 24 07:51:18.227546 r-leopard-42 INFO lldp#supervisord 2024-05-24 04:51:18,227 INFO waiting for supervisor-proc-exit-listener, rsyslogd, lldpd, lldp-syncd, lldpmgrd to die
2024 May 24 07:51:18.227546 r-leopard-42 ERR pmon#stormond[53]: Unable to sync state_db to disk: 'sda'
2024 May 24 07:51:18.227562 r-leopard-42 INFO pmon#stormond[53]: Exiting with SIGTERM
2024 May 24 07:51:18.227644 r-leopard-42 INFO pmon#stormond[53]: Shutting down Storage Monitoring Daemon
2024 May 24 07:51:18.303392 r-leopard-42 INFO pmon#supervisord 2024-05-24 04:51:18,302 WARN stopped: stormond (exit status 143)
2024 May 24 07:51:18.303417 r-leopard-42 INFO pmon#syseepromd[51]: Caught signal 'SIGTERM' - exiting...
```

Reply:
The "Unable to sync state_db to disk: 'sda'" bug is because of an incorrect dict initialization and the fix has been pushed upstream. That said, I found a couple edge case bugs during testing yesterday. I will fix those too and push them upstream shortly.

Reply:
Fixed in latest